test/index.test.js

```diff
@@ -0,0 +1,1045 @@
+'use strict';
+
+/* eslint-disable no-underscore-dangle */
+
+const chai = require('chai');
+const assert = chai.assert;
+const mockery = require('mockery');
+const sinon = require('sinon');
+const testScm = require('./data/testScm');
+// const sinonStubPromise = require('sinon-stub-promise');
+
+sinon.assert.expose(chai.assert, { prefix: '' });
+// sinonStubPromise(sinon);
+
+describe('index test', () => {
+    let Scm;
+    let scm;
+    let githubScmMock;
+    let gitlabScmMock;
+    let exampleScmMock;
+    const ecosystem = {
+        api: 'http://api.com',
+        store: 'http://store.com'
+    };
+    const githubPluginOptions = {
+        displayName: 'github.com',
+        oauthClientId: 'OAUTH_CLIENT_ID',
+        oauthClientSecret: 'OAUTH_CLIENT_SECRET',
+        username: 'USERNAME',
+        email: 'EMAIL',
+        secret: 'SECRET',
+        privateRepo: false
+    };
+    const gitlabPluginOptions = {
+        displayName: 'gitlab.com',
+        privateRepo: true
+    };
+    const examplePluginOptions = {
+        displayName: 'example.com',
+        somekey: 'somevalue'
+    };
+    const initMock = (mock, plugin) => {
+        [
+            'dummyFunction',
+            'dummyRejectFunction',
+            'addWebhook',
+            'parseUrl',
+            'parseHook',
+            'getCheckoutCommand',
+            'decorateUrl',
+            'decorateCommit',
+            'decorateAuthor',
+            'getPermissions',
+            'getCommitSha',
+            'updateCommitStatus',
+            'getFile',
+            'getOpenedPRs',
+            'getPrInfo'
+        ].forEach((method) => {
+            mock[method].resolves(plugin);
+        });
+        mock.getBellConfiguration.resolves({ [plugin]: `${plugin}Bell` });
+        mock.stats.returns({ [plugin]: { requests: plugin } });
+        mock.canHandleWebhook.resolves(true);
+        mock.getScmContexts.returns([`${plugin}.context`]);
+        mock.getDisplayName.returns(plugin);
+    };
+
+    before(() => {
+        mockery.enable({
+            useCleanCache: true,
+            warnOnUnregistered: false
+        });
+    });
+
+    beforeEach(() => {
+        githubScmMock = {
+            dummyFunction: sinon.stub(),
```

Comment:
This part could all be added to the initMock  method, too.

```
mock[method] = sinon.stub().resolves(plugin)
```

This makes the setup really clean:
```
githubScmMock = initMock('github');
```